packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -942,10 +942,9 @@ class LocalizationsGenerator {
   @visibleForTesting
   static File templateArbFileFromFileName(String templateArbFileName, Directory inputDirectory) {
     final File templateArbFile = inputDirectory.childFile(templateArbFileName);
-    final FileStat templateArbFileStat = templateArbFile.statSync();
-    final String templateArbFileStatModeString = templateArbFileStat.modeString();
-    final FileSystemEntityType templateArbFileStatType = templateArbFileStat.type;
-    if (templateArbFileStatType == FileSystemEntityType.notFound) {
+    final String templateArbFileStatModeString = templateArbFile.statSync().modeString();
+    final bool templateArbFileExists = templateArbFile.existsSync();
+    if (!templateArbFileExists) {
```

Comment:
Could just be `if (!templateArbFile.existsSync())`

Reply:
Sorry, I was looking at the diffs not at the overall context. The version of the code you had originally, that just relied on one stat call made sense as it was. Sorry about the (my) confusion.